src/coverlet.core/Symbols/CecilSymbolHelper.cs

```diff
@@ -19,14 +19,29 @@ internal static class CecilSymbolHelper
     {
         private const int StepOverLineCode = 0xFEEFEE;
 
+        private static bool IsCompilerGenerated(MethodDefinition methodDefinition)
```

Comment:
Can you add a comment like `// In case of nested state machines CompilerGeneratedAttribute is present only on outermost generated class so we traverse parent/child chain botton-top`